electrumx/server/http_session.py

```diff
@@ -323,55 +325,45 @@ async def transaction_get(self, tx_hash, verbose=False):
+            raise Exception(f'"verbose" must be a boolean')
 
         return await self.daemon_request('getrawtransaction', tx_hash, verbose)
-    
-    async def transaction_get_detail(self, tx_hash):
-        '''
-        Return the serialized raw transaction given its hash\n
-        tx_hash: the transaction hash as a hexadecimal string
-        verbose: passed on to the daemon
-        '''
-        self.assert_tx_hash(tx_hash)
-        return await self.daemon_request('getrawtransaction', tx_hash, True)
-    
-    async def transaction_get_multiple(self, tx_hashes):
-        '''
-        Return the a bunch of serialized raw transaction given hash values\n
-        tx_hashes: list of transaction hash;
-        verbose: passed on to the daemon;
-        '''
-        return await self.daemon_request('getrawtransactions', tx_hashes)
-    
-    async def transaction_get_multiple_details(self, tx_hashes):
-        '''
-        Return the a bunch of serialized raw transaction given hash values\n
-        tx_hashes: list of transaction hash;
-        verbose: passed on to the daemon;
-        '''
-        return await self.daemon_request('getdetailedtransactions', tx_hashes)
-
-    async def descriptorinfo_get_multiple(self, descriptors, script_type):
-        return await self.daemon_request('getdescriptorsinfo', descriptors, script_type)
-        
-    async def address_get_multiple(self, descriptors):
-        return await self.daemon_request('getderiveaddresses', descriptors)
-    
-    async def mempool_get(self, verbose=False):
-        '''Returns all transaction ids in memory pool as a json array of string transaction ids
-
-        verbose: True for a json object, false for array of transaction ids
-        '''
-        if verbose not in (True, False):
-            raise RPCError(BAD_REQUEST, f'"verbose" must be a boolean')
 
-        return await self.daemon_request('getrawmempool', verbose)
+    # new daemon calls
+    async def get_tx_raw_list(self, txid_list):
+        for txid in txid_list:
+            if not(txid and isinstance(txid, str) and len(txid)==64):
+                raise Exception('invalid format of txid as argument')
+        return await self.daemon_request('getrawtransactions', txid_list)
+
+    async def get_tx_detail_list(self, txid_list):
+        for txid in txid_list:
+            if not(txid and isinstance(txid, str) and len(txid)==64):
+                raise Exception('invalid format of txid as argument')
+        return await self.daemon_request('getdetailedtransactions', txid_list)
 
+    async def get_script_detail_list(self, script_list):
+        return await self.daemon_request('decode_scripts', script_list)
 
 class DecimalEncoder(json.JSONEncoder):
     def default(self, o):
         if isinstance(o, Decimal):
             return float(o)
         return super(DecimalEncoder, self).default(o)
 
+def logging_middleware(self) -> web_middlewares:
```

Comment:
is the logging format be finalized version?

Reply:
No, just a rough format. Not sure if we have a formal log format.